Docker build failure with cuDNN

As reported by @cobie8a in #63:

```
Step 22 : RUN luarocks install https://raw.githubusercontent.com/jcjohnson/torch-rnn/master/torch-rnn-scm-1.rockspec
---> Running in 1fa7d40d22f5
Cloning into 'torch-rnn'...
Using https://raw.githubusercontent.com/jcjohnson/torch-rnn/master/torch-rnn-scm-1.rockspec... switching to 'build' mode
Updating manifest for /root/torch/install/lib/luarocks/rocks
torch-rnn scm-1 is now built and installed in /root/torch/install/ (license: MIT)

---> debdad6a31b9
Removing intermediate container 1fa7d40d22f5
Step 23 : RUN luarocks install loadcaffe
---> Running in 49ea1a7add60
Cloning into 'loadcaffe'...
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Torch7 in /root/torch/install
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE

-- Found PROTOBUF: /usr/lib/x86_64-linux-gnu/libprotobuf.so

-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/luarocks_loadcaffe-1.0-0-9452/loadcaffe/build
[ 33%] Running C++ protocol buffer compiler on caffe.proto
Scanning dependencies of target loadcaffe
[ 66%] Building CXX object CMakeFiles/loadcaffe.dir/loadcaffe.cpp.o
[100%] Building CXX object CMakeFiles/loadcaffe.dir/caffe.pb.cc.o
Linking CXX shared module libloadcaffe.so
[100%] Built target loadcaffe
[100%] Built target loadcaffe
Install the project...
-- Install configuration: "Release"
-- Installing: /root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lib/libloadcaffe.so
-- Set runtime path of "/root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lib/libloadcaffe.so" to "$ORIGIN/../lib:/root/torch/install/lib"
-- Installing: /root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lua/loadcaffe/loadcaffe.lua
-- Installing: /root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lua/loadcaffe/ffi.lua
-- Installing: /root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lua/loadcaffe/test.lua
-- Installing: /root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0/lua/loadcaffe/init.lua
Installing https://raw.githubusercontent.com/torch/rocks/master/loadcaffe-1.0-0.rockspec...
Using https://raw.githubusercontent.com/torch/rocks/master/loadcaffe-1.0-0.rockspec... switching to 'build' mode
cmake -E make_directory build;
cd build;
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/root/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/root/torch/install/lib/luarocks/rocks/loadcaffe/1.0-0"; 
make

cd build && make install
Updating manifest for /root/torch/install/lib/luarocks/rocks
loadcaffe 1.0-0 is now built and installed in /root/torch/install/ (license: BSD)

---> b90f8ba3237d
Removing intermediate container 49ea1a7add60
Step 24 : RUN mkdir /nvidia_cudnn
---> Running in 583bd371832c
---> 5e88757a262c
Removing intermediate container 583bd371832c
Step 25 : ADD extras/cudnn-7.5-linux-x64-v5.1-rc.tgz /nvidia_cudnn
lstat extras/cudnn-7.5-linux-x64-v5.1-rc.tgz: no such file or directory
make: *** [build] Error 1
```

@ymt123 suggests the solution is to install cuDNN with the following code:

```
# Install cuDNN
RUN echo "deb http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1404/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list
ENV CUDNN_VERSION 5
LABEL com.nvidia.cudnn.version="5"
RUN apt-get update && apt-get install -y --no-install-recommends --force-yes \
       libcudnn5 \
       libcudnn5-dev && \
    rm -rf /var/lib/apt/lists/*
```

--- a/Dockerfile.densecap
+++ b/Dockerfile.densecap
@@ -1,4 +1,3 @@
-
 # Be sure to download cuda from NVIDIA first before building this
 # container. NVIDIA requires users to register as a developer. Afterwards,
 # you can download the .tgz file. 
@@ -56,12 +55,14 @@ RUN luarocks install https://raw.githubusercontent.com/qassemoquab/stnbhwd/maste
 RUN luarocks install https://raw.githubusercontent.com/jcjohnson/torch-rnn/master/torch-rnn-scm-1.rockspec
 RUN luarocks install loadcaffe
 
-# Add NVIDIA's cuDNN 
-RUN mkdir /nvidia_cudnn
-ADD extras/cudnn-7.5-linux-x64-v5.1-rc.tgz /nvidia_cudnn
-RUN sudo cp -P /nvidia_cudnn/cuda/include/cudnn.h /usr/include
-RUN sudo cp -P /nvidia_cudnn/cuda/lib64/libcudnn* /usr/lib/x86_64-linux-gnu/
-RUN sudo chmod a+r /usr/lib/x86_64-linux-gnu/libcudnn*
+# Install cuDNN
+RUN echo "deb http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1404/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list
+ENV CUDNN_VERSION 5
+LABEL com.nvidia.cudnn.version="5"
+RUN apt-get update && apt-get install -y --no-install-recommends --force-yes \
+        libcudnn5 \
+        libcudnn5-dev && \
+    rm -rf /var/lib/apt/lists/*
 
 # GPU acceleratation
 RUN luarocks install cutorch

--- a/Dockerfile.densecap
+++ b/Dockerfile.densecap
@@ -1,31 +1,29 @@
-# Be sure to download cuda from NVIDIA first before building this
-# container. NVIDIA requires users to register as a developer. Afterwards,
-# you can download the .tgz file. 
-
 FROM  l41-nvidia-base
+
 # Install git, apt-add-repository and dependencies for iTorch
 RUN apt-get update && apt-get install -y \
-  git \
-  software-properties-common \
-  ipython3 \
-  libssl-dev \
-  libzmq3-dev \
-  python-zmq \
-  python-pip \
-  libhdf5-serial-dev \
-  hdf5-tools \
-  libprotobuf-dev \ 
-  protobuf-compiler \
-  default-jre
+    git \
+    software-properties-common \
+    ipython3 \
+    libssl-dev \
+    libzmq3-dev \
+    python-zmq \
+    python-pip \
+    libhdf5-serial-dev \
+    hdf5-tools \
+    libprotobuf-dev \
+    protobuf-compiler \
+    default-jre
 
 # Install Jupyter Notebook for iTorch
 RUN pip install notebook ipywidgets
 
 # Run Torch7 installation scripts (dependencies only)
-RUN git clone https://github.com/torch/distro.git /root/torch --recursive && cd /root/torch && \
-  bash install-deps
+RUN git clone https://github.com/torch/distro.git /root/torch --recursive && \
+    cd /root/torch && \
+    bash install-deps
 RUN cd /root/torch && \
-  ./install.sh
+    ./install.sh
 
 # Export environment variables manually
 ENV LUA_PATH='/root/.luarocks/share/lua/5.1/?.lua;/root/.luarocks/share/lua/5.1/?/init.lua;/root/torch/install/share/lua/5.1/?.lua;/root/torch/install/share/lua/5.1/?/init.lua;./?.lua;/root/torch/install/share/luajit-2.1.0-beta1/?.lua;/usr/local/share/lua/5.1/?.lua;/usr/local/share/lua/5.1/?/init.lua'
@@ -37,14 +35,17 @@ ENV LUA_CPATH='/root/torch/install/lib/?.so;'$LUA_CPATH
 
 # Install cudnn manually as Torch opts for latest version
 #RUN luarocks install https://raw.githubusercontent.com/soumith/cudnn.torch/R4/cudnn-scm-1.rockspec
-RUN cd /root && git clone https://github.com/deepmind/torch-hdf5.git && \
-  cd torch-hdf5 && luarocks make hdf5-0-0.rockspec LIBHDF5_LIBDIR="/usr/lib/x86_64-linux-gnu/"
+RUN cd /root && \
+    git clone https://github.com/deepmind/torch-hdf5.git && \
+    cd torch-hdf5 && \
+    luarocks make hdf5-0-0.rockspec LIBHDF5_LIBDIR="/usr/lib/x86_64-linux-gnu/"
 
 # add FindCUDA to remove annoying warnings
-RUN wget https://github.com/hughperkins/FindCUDA/archive/v3.5-1.tar.gz -q -O FindCUDA-v3.5-1.tar.gz 
-RUN tar -zxf FindCUDA-v3.5-1.tar.gz
-RUN cd FindCUDA-3.5-1 && luarocks make rocks/findcuda-scm-1.rockspec
-RUN rm /FindCUDA-v3.5-1.tar.gz
+RUN wget https://github.com/hughperkins/FindCUDA/archive/v3.5-1.tar.gz -q -O FindCUDA-v3.5-1.tar.gz && \
+    tar -zxf FindCUDA-v3.5-1.tar.gz && \
+    cd FindCUDA-3.5-1 && \
+    luarocks make rocks/findcuda-scm-1.rockspec && \
+    rm /FindCUDA-v3.5-1.tar.gz
 
 # Install additional DenseCap dependencies
 RUN luarocks install torch

